packages/fxa-auth-server/lib/oauth/jwt_id_token.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AppError = require('./error');
+const jwt = require('./jwt');
+
+const HEADER_TYP = 'at+JWT';
+
+/**
+ * Verify an OIDC ID Token, following the flow in 3.1.3.7 of the spec:
+ * https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
+ *
+ * @param {String} idToken
+ * @param {String} clientId
+ * @throws `invalidToken` error if ID Token is invalid.
+ * @returns {Promise<Object>} resolves with claims in Token
+ */
```

Comment:
To help set expectations for future readers, I think it would be worth saying a few words at the top of this function to explain that it does all the checks that an RP would ideally perform locally; that is, that it's operating *from the perspective on an RP* which is a slightly unusual thing for code in this repo to be doing.

(I know that's what you're doing and it took me a little while for that to really "click" when reading the code here, so it's probably good context for other readers).